androidApp/src/main/java/press/home/HomeView.kt

```diff
@@ -113,10 +124,58 @@ class HomeView @AssistedInject constructor(
     )
   }
 
+  private val createEditorView = { noteUuid: Uuid ->
+    editorViewFactory.create(
+        context = context,
+        openMode = ExistingNote(noteUuid),
+        onDismiss = { notesList.collapse() }
+    )
+  }
+
+  private var noteForEditor by observable<NoteParcelable?>(null) { _, oldNote, newNote ->
+    if (oldNote == newNote) return@observable
+    if (newNote != null) {
+      val editorView = createEditorView(requireNotNull(Uuid.parse(newNote.noteUuid)))
+      editorView.id = R.id.editorViewId
+      noteEditorPage.addView(editorView)
+      noteEditorPage.doOnNextCollapse { it.removeView(editorView) }
+
+      val keyboardToggle = HideKeyboardOnPageCollapse(editorView.editorEditText)
+      noteEditorPage.addStateChangeCallbacks(keyboardToggle)
+      noteEditorPage.doOnNextCollapse { it.removeStateChangeCallbacks(keyboardToggle) }
+
+      noteEditorPage.pullToCollapseInterceptor =
+        interceptIfViewCanBeScrolled(editorView.scrollView)
+
+      noteEditorPage.post {
+        notesList.expandItem(itemId = newNote.adapterId, immediate = false)
+      }
+    } else {
+      notesList.collapse()
+    }
+  }
+
   init {
     setupNoteEditorPage()
   }
 
+  override fun onSaveInstanceState(): Parcelable? {
+    val bundle = Bundle()
+    val superState = super.onSaveInstanceState()
+    if (superState != null) bundle.putParcelable(KEY_SUPER, superState)
+    noteForEditor?.let { bundle.putParcelable(KEY_NOTE_MODEL, it) }
+    return bundle
```

Comment:
Let's use a Parcelable data class for saving HomeView's entire state instead of just `NoteParcelable`. It'll remove the need of maintaining string keys and will add type safety. 

I imagine it'll look something like this:

```kotlin
override fun onSaveInstanceState(): Parcelable {
  return HomeViewSavedState(
    super = super.onSaveInstanceState(),
    openNote = openNote?.let { HomeViewSavedState.OpenNote(it.uuid, it.adapterId) }
  )
}

override fun onRestoreInstanceState(state: Parcelable) {
  require(state is HomeViewSavedState)
  noteForEditor = state.openNote
  super.onRestoreInstanceState(state.super)
}
```

Reply:
Sounds good. I wrote this before reading your implementation of InboxRecyclerView which looks better. Will update real quick.